Assets/Scripts/Environment/Hangar/Battery.cs

```diff
@@ -55,38 +62,77 @@ private void SetEmission()
         meshRenderer.material.SetColor("_EmissionColor", emission);
     }
 
-    public bool PlayerIsHolding()
+    /*public bool PlayerIsHolding()
     {
-        return PlayerManager.PlayerObject.GetComponentInChildren<Battery>() != null;
+        return springJoint != null && springJoint.connectedBody == PlayerManager.PlayerMovement.PlayerRigidbody;
+
+        return PlayerManager.IsHoldingBattery;
     }
+    */
 
     public void TakeBattery()
     {
-        Container.ParentToPlayer();
+        Container.transform.localPosition = new Vector3(Container.transform.localPosition.x, HangarConstants.BatteryYPosition, Container.transform.localPosition.z);
+
+        containerRigidBody.useGravity = false;
+
+        // We add constraints so that the battery doesn't swing around uncontrollably
+        containerRigidBody.constraints = HangarConstants.BatteryRigidbodyConstraints;
+
+        // Setting the container's spring joint values
+        springJoint = Container.AddComponent<SpringJoint>();

```

Comment:
Max has the rest of the code covered, so I'm going to be really nit picky. AddComponent is a relatively expensive call, so you could just add the component to the prefab, and disable it. Then here all you'd have to do is enable the component

Reply:
AFAIK you can't disable spring joints. Like rigid bodies, they have to be destroyed. I think if you set the connected body to null the joint is still connected to some point in world space. Maybe there's another option though.